Confus/Confus/LocalPlayerController.cpp

```diff
@@ -16,30 +16,27 @@ namespace Confus
 
 	void LocalPlayerController::handleInput(ConfusShared::EventManager& a_EventManager)
 	{
-		m_InputState.ForwardPressed = m_InputState.ForwardPressed || a_EventManager.IsKeyDown(irr::EKEY_CODE::KEY_KEY_W);
-		m_InputState.BackwardPressed = m_InputState.BackwardPressed || a_EventManager.IsKeyDown(irr::EKEY_CODE::KEY_KEY_S);
-		m_InputState.LeftPressed = m_InputState.LeftPressed || a_EventManager.IsKeyDown(irr::EKEY_CODE::KEY_KEY_A);
-		m_InputState.RightPressed = m_InputState.RightPressed || a_EventManager.IsKeyDown(irr::EKEY_CODE::KEY_KEY_D);
-		m_InputState.LeftMouseButtonPressed = m_InputState.LeftMouseButtonPressed || a_EventManager.IsLeftMouseDown();
-		m_InputState.RightMouseButtonPressed = m_InputState.RightMouseButtonPressed || a_EventManager.IsRightMouseDown();
-
-		if(a_EventManager.IsRightMouseDown())
-		{
-			m_Player.startHeavyAttack();
-		}
-		else if(a_EventManager.IsLeftMouseDown())
-		{
-			m_Player.startLightAttack();
-		};
+        if (m_Player.getState() != ConfusShared::EPlayerState::HeavyAttacking && m_Player.getState() != ConfusShared::EPlayerState::LightAttacking)
+        {
+            m_InputState.ForwardPressed = m_InputState.ForwardPressed || a_EventManager.IsKeyDown(irr::EKEY_CODE::KEY_KEY_W);
+            m_InputState.BackwardPressed = m_InputState.BackwardPressed || a_EventManager.IsKeyDown(irr::EKEY_CODE::KEY_KEY_S);
+            m_InputState.LeftPressed = m_InputState.LeftPressed || a_EventManager.IsKeyDown(irr::EKEY_CODE::KEY_KEY_A);
+            m_InputState.RightPressed = m_InputState.RightPressed || a_EventManager.IsKeyDown(irr::EKEY_CODE::KEY_KEY_D);
+            m_InputState.LeftMouseButtonPressed = m_InputState.LeftMouseButtonPressed || a_EventManager.IsLeftMouseDown();
+            m_InputState.RightMouseButtonPressed = m_InputState.RightMouseButtonPressed || a_EventManager.IsRightMouseDown();
+        } 
+        else
+        {
+            resetInputState();
```

Comment:
I guess it's fine, but it seems a bit weird to me that your input for walking will be dropped as soon as you attack.